api/v1beta1/clusterclass_types.go

```diff
@@ -158,9 +158,42 @@ type VariableSchema struct {
 // which are not supported in CAPI have been removed.
 type JSONSchemaProps struct {
 	// Type is the type of the variable.
-	// Valid values are: string, integer, number or boolean.
+	// Valid values are: object, array, string, integer, number or boolean.
 	Type string `json:"type"`
 
+	// Properties specifies fields of an object.
```

Comment:
NOTE: depending on merge order with https://github.com/kubernetes-sigs/cluster-api/pull/5752 we have to add those fields to the `schemasAreEqual` funcs

Reply:
Not necessary anymore as we're now using reflect.DeepEqual